### PR TITLE
User can create a post

### PR DESCRIPTION
**Pull request by KyunginNa**
[User can create a post with a song attached](https://www.pivotaltracker.com/story/show/176696241)
## User Story
```
As a User,            
In order to share my taste in music,         
I want to be able to create a new post with a song attached.
```
## Changes proposed in this pull request:
* Creating a test to check if the user can create a post
* Adding Card for track preview in PostForm component
* Creating create a function in the PostService module 
* Modifying the PostForm jest test 

## What I have learned working on this feature:
* How to find an error that is not originated from the HTTP request

## Screenshots (Client)
Screenshot 2021-01-29 at 17 02 42

### CODE REVIEW

File: src/components/PostForm.jsx [hunk 1]
```diff
@@ -1,37 +1,86 @@
...
+            testID='postButton'
+            title='Post'
+            color='black'
+            onPress={() => PostService.create(trackDetails, description, props.navigation.navigate)}
+          />
+        </>
+      )}
+      <Text testID='errorMessage'>
+        {errorMessage}
+      </Text>
```
Comment (sealfury) [inline src/components/PostForm.jsx:83]: I'm not convinced that this is the best way of rendering the preview card but as a whole this component is quite impressive so we can put that aside for now :)

**Review by sealfury — APPROVED** [review]

File: src/components/PostForm.jsx [hunk 2]
```diff
@@ -1,37 +1,86 @@
...
-        onChangeText={text => setSearch(text)}
+        placeholder='Search a song to create a post!'
+        onChangeText={(text) => setSearch(text)}
         value={search || ''}
       />
       <Button
+        style={styles.searchButton}
         testID='searchButton'
         title='Search'
+        color='black'
```
Comment (ViktorHek) [inline src/components/PostForm.jsx:29]: you should be able to change the color of the button in the styles component `searchButton`

File: src/components/PostForm.jsx [hunk 3]
```diff
@@ -1,37 +1,86 @@
...
+            style={styles.postDescription}
+            placeholder='Write a caption!'
+            testID='descriptionInput'
+            onChangeText={(text) => setDescription(text)}
+          />
+          <Button
+            style={styles.postButton}
+            testID='postButton'
+            title='Post'
+            color='black'
```
Comment (ViktorHek) [inline src/components/PostForm.jsx:76]: you should be able to change the color of the button in the styles component `searchButton`

File: src/components/PostForm.jsx (hunk 2 above)
Comment (KyunginNa) [reply src/components/PostForm.jsx:29]: We tried putting `color` and `background color` in the style sheet, but both didn't work unfortunately...

**Pull request merged**